common_code/inspiral/CE_orbit.f90

```diff
@@ -173,14 +294,32 @@ subroutine check_merger(id, ierr)
          CE_companion_radius = s% xtra(3)
          CE_companion_mass = s% xtra(4)
 
-         rl_core = eval_rlobe(s% he_core_mass,CE_companion_mass,CE_companion_position)*CE_companion_position
-         rl_companion = eval_rlobe(CE_companion_mass,s% he_core_mass,CE_companion_position)*CE_companion_position
+         CE_companion_molec_weight = s% x_ctrl(19)
+
+         rl_core = eval_rlobe(s% he_core_mass,CE_companion_mass,CE_companion_position)
+         rl_companion = eval_rlobe(CE_companion_mass,s% he_core_mass,CE_companion_position)
```

Comment:
Isn't this causing issues, when the `s% he_core_mass` is not a well defined positive value, e.g. on the MS of (super) massive stars?